include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -355,11 +355,20 @@ class BaseReduceMinLoc
   constexpr BaseReduceMinLoc() : Base(value_type(T(), IndexType())) {}
 
   constexpr BaseReduceMinLoc(T init_val, IndexType init_idx,
-                             T identity_ = reduce_type::identity())
-    : Base(value_type(init_val, init_idx), identity_)
+                             T identity_val_ = reduce_type::identity(),
+                             IndexType identity_loc_ = DefaultLoc<IndexType>().value())
+    : Base(value_type(init_val, init_idx), value_type(identity_val_, identity_loc_))
   {
   }
 
+  void reset(T init_val, IndexType init_idx = DefaultLoc<IndexType>().value(),
```

Comment:
Thanks for catching these hidden problems! For the loc reductions, should we require the `init_idx` as a second reset argument to distinguish it from a normal non-loc reduction?

Reply:
I think that's a good idea but it would be a change.

Reply:
Yeah, it would change the interface subtly. I'm fine with we have now too.

Reply:
How about we leave it as it was for now, then change it for the next release?

Reply:
> How about we leave it as it was for now, then change it for the next release?

Sounds good to me.